File: solutions/0121-best-time-to-buy-and-sell-stock/solution.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        # min_price= float('inf')
        # max_profit=0

        # for price in prices:
        #     if price< min_price:
        #         min_price= price
        #     else:
        #         max_profit= max(max_profit, price-min_price)
        # return max_profit

        maxProfit=0
        bestBuy = prices[0]
        for i in range(1, len(prices)):
            if prices[i]> bestBuy:
                maxProfit = max(maxProfit, prices[i]-bestBuy)
        
            bestBuy = min(bestBuy, prices[i])
        return maxProfit

File: solutions/0121-best-time-to-buy-and-sell-stock/solution.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let maxProfit =0;
    let buy= 0, sell=1;

    while(sell < prices.length){
        if(prices[sell] > prices[buy]){
            maxProfit = Math.max(maxProfit, prices[sell]-prices[buy]);
        }
        else{
            buy = sell;
        }
        sell++;
    }

return maxProfit;







    // let profit = 0;
    // let maxProfit = 0;
    // for(let i=0; i< prices.length; i++){
    //     for(let j=i+1; j< prices.length; j++){
    //         profit = prices[j]- prices[i];
    //         if (profit > 0){
    //             maxProfit = Math.max(maxProfit, profit);
    //     }
    //     }
        
    // }
    // return maxProfit;
};
